lib/improver/cli/nbhood.py

```diff
@@ -206,74 +206,157 @@ def main(argv=None):
         if args.apply_recursive_filter:
             parser.error('Cannot process complex numbers with recursive '
                          'filter')
+    mask_cube = None
+    if args.input_mask_filepath:
+        mask_cube = load_cube(args.input_mask_filepath)
 
+    alphas_x_cube = None
+    alphas_y_cube = None
+    if args.input_filepath_alphas_x_cube is not None:
+        alphas_x_cube = load_cube(args.input_filepath_alphas_x_cube)
+    if args.input_filepath_alphas_y_cube is not None:
+        alphas_y_cube = load_cube(args.input_filepath_alphas_y_cube)
+
+    # Load Cube
     cube = load_cube(args.input_filepath)
-    if args.degrees_as_complex:
+
+    # Process Cube
+    result = process(cube, args.neighbourhood_output,
+                     args.neighbourhood_shape, args.radius,
+                     args.radii_by_lead_time, args.degrees_as_complex,
+                     args.weighted_mode, args.sum_or_fraction, args.re_mask,
+                     args.percentiles, mask_cube, args.halo_radius,
+                     args.apply_recursive_filter, alphas_x_cube,
+                     alphas_y_cube, args.alpha_x, args.alpha_y,
+                     args.iterations)
+
+    # Save Cube
+    save_netcdf(result, args.output_filepath)
+
+
+def process(cube, neighbourhood_output, neighbourhood_shape, radius,
+            radii_by_lead_time, degrees_as_complex=False,
+            weighted_mode=False, sum_or_fraction="fraction", re_mask=False,
+            percentiles=DEFAULT_PERCENTILES, mask_cube=None,
+            halo_radius=None, apply_recursive_filter=False, alphas_x_cube=None,
+            alphas_y_cube=None, alpha_x=None, alpha_y=None, iterations=1):
```

Comment:
Seeing a function signature with this many keyword arguments makes me wonder. We are effectively duplicating handling of default values - once in the argparser and second time in the `process` functions. This is probably not a good idea (there's nothing stopping them to go out of sync in the future). Not sure how to best handle this yet. @LaurenceBeard, @arh89, any thoughts?

Reply:
There is a different problem as well with the complexity of the nbhood interfaces - I'm hoping to simplify that when I can and remove a lot of unused functionality.

Reply:
Yes, but that's a separate issue. I'm thinking more about getting the pattern right in the general case rather than this particular one. Perhaps let's ask the question, what's the long term plan for CLIs? Do we keep maintaining them going forward? If not, then this solution is fine in the interim period of phasing them out. But if we do want to keep CLIs, however, then we should think how to manage this duplication of interfaces.

Reply:
Yes, we really want one to fall out of the other.
eg. A primary interface (the functional form) and a second that you get for free (the cli)
(..in the case that we have interfaces that call multiple plugins we could have a functional representation of that.)
Either should use something like python-fire to achieve this, or extend improver/argparser.py to do just this.

Reply:
I agree with both @LaurenceBeard and @TomekTrzeciak. For now, we'll use what Neil has implemented, and longer term, if we intend on keeping the CLIs then we should generate them from the function signature (or from a plugin class...). To be honest, I don't see a big advantage to keeping the CLIs, given all of the discussions we've been having about infrastructure recently.

The next step for this work, in my opinion, would be to review the `process` functions that Neil has created, and figuring out how much of that functionality should be migrated to plugin classes. If we keep the CLIs, they definitely need to follow a `load`-`plugin`-`save` pattern, with that logic centralised as much as possible (into one file).